Engine/AlgorithmManager.cs

```diff
@@ -1257,16 +1239,42 @@ private void ProcessSplitSymbols(IAlgorithm algorithm, List<Split> splitWarnings
         /// <summary>
         /// Samples the benchmark in a  try/catch block
         /// </summary>
-        private void SampleBenchmark(IAlgorithm algorithm, IResultHandler results, DateTime time)
+        private void SampleBenchmark(IResultHandler results, DateTime time)
         {
             try
             {
                 // backtest mode, sample benchmark on day changes
-                results.SampleBenchmark(time, algorithm.Benchmark.Evaluate(time).SmartRounding());
+                results.SampleBenchmark(time, _algorithm.Benchmark.Evaluate(time).SmartRounding());
+            }
+            catch (Exception err)
+            {
+                _algorithm.RunTimeError = err;
+                _algorithm.Status = AlgorithmStatus.RuntimeError;
+                Log.Error(err);
+            }
+        }
+
+        /// <summary>
+        /// Samples the performance in a  try/catch block and update the <see cref="_dailyPortfolioValue"/>
+        /// </summary>
+        private void SamplePerformance(IResultHandler results, DateTime time, bool force = false)
+        {
+            try
+            {
+                if (_previousTime.Date != time.Date || force)
+                {
+                    var currentPortfolioValue = _algorithm.Portfolio.TotalPortfolioValue;
+
+                    results.SamplePerformance(_previousTime.Date, _dailyPortfolioValue == 0m ? 0
```

Comment:
Note for posterity. This was sampling 0 before to ensure the array of data for benchmark and the equity array were the same length -- otherwise, beta/correlation numbers swing because the data is offset by 1